### PR TITLE
Make post modal show up instantaneously

### PR DESCRIPTION
**Pull request by YLivay**

### CODE REVIEW

File: src/app/components/post/add-button/add-button.ts [hunk 1]
```diff
@@ -55,11 +55,12 @@ export default class AppPostAddButton extends Vue {
...
-			this.game ? this.game.id : 0
-		);
+		const postProvider = FiresidePost.$create(this.game ? this.game.id : 0);
 
-		post = await PostEditModal.show(post, { community: this.community, channel: this.channel });
+		const post = await PostEditModal.show(postProvider, {
+			community: this.community,
+			channel: this.channel,
+		});
 		this._isBlocked = false;
```
Comment (nilllzz) [inline src/app/components/post/add-button/add-button.ts]: Remove the isBlocked stuff, not needed when the modal opens right away.

File: src/app/components/forms/post/post.styl [hunk 2]
```diff
@@ -124,6 +124,9 @@
 			border-top: 1px solid var(--theme-bg-subtle)
 			border-bottom: 1px solid var(--theme-bg-subtle)
 
+			> :not(:first-child)
```
Comment (nilllzz) [inline src/app/components/forms/post/post.styl]: What is this?

File: src/app/components/post/edit-modal/edit-modal.ts [hunk 3]
```diff
@@ -1,25 +1,40 @@
...
 	components: {
 		FormPost,
+		AppPostAddPlaceholder,
+		AppLoadingFade,
 	},
 })
 export default class AppPostEditModal extends BaseModal {
-	@Prop(FiresidePost)
-	post!: FiresidePost;
+	@Prop(Object)
```
Comment (nilllzz) [inline src/app/components/post/edit-modal/edit-modal.ts]: `Object` doesn't cover `Promise`, this needs to be `undefined`.

File: src/app/components/post/edit-modal/edit-modal.vue [hunk 4]
```diff
@@ -7,12 +7,16 @@
...
 
 		<div class="modal-body">
-			<form-post
-				:model="post"
-				:default-community="community"
-				:default-channel="channel"
-				@submit="onSubmitted"
-			/>
+			<app-loading-fade :is-loading="!post">
+				<app-post-add-placeholder v-if="!post" />
```
Comment (nilllzz) [inline src/app/components/post/edit-modal/edit-modal.vue]: Should probably make a isLoading getter and check here.

File: src/app/components/user/spawn-day/spawn-day.ts [hunk 5]
```diff
@@ -84,16 +84,17 @@ export default class AppUserSpawnDay extends Vue {
...
-		writer.appendTag('spawnday');
+			newPost.lead_content = spawnDayDoc.toJson();
+			return newPost;
+		});
 
-		post.lead_content = spawnDayDoc.toJson();
-
-		post = await PostEditModal.show(post);
+		const post = await PostEditModal.show(postProvider);
 		this._isBlocked = false;
```
Comment (nilllzz) [inline src/app/components/user/spawn-day/spawn-day.ts]: Remove isBlocked here too.

File: src/app/components/forms/post/post.styl (hunk 2 above)
Comment (YLivay) [reply src/app/components/forms/post/post.styl]: On mobile it turns out the buttons dont have any margins between them so it's easy to mispress. This fixes it.

Comment (nilllzz) [reply src/app/components/forms/post/post.styl]: Gotcha.

**Pull request merged**